server/client.go

```diff
@@ -447,6 +447,12 @@ func (c *client) processConnect(arg []byte) error {
 			srv.mu.Unlock()
 		}
 
+		// Check for max connections
```

Comment:
This is what I did first (hence the fix in `clearConnection` because when calling `closeConnection` before client is initialized would crash since `c.bw` was `nil`).
Keep in mind that the client expects an INFO before any possible error message. So I need to do the send after the client is initialized (buffer created so that sendErr actually works) and after sending the server info, all that in `createClient()`. The problem with this approach is that the client will get the error only, in my tests, 10% of the cases. Otherwise, it either gets `EOF` or `connection reset by peer`. With the original approach, it gets the error 100% of the times. That being said, I understand the argument of telnet, etc.. so I guess it is worth having the client not always know why it failed to connect but maximizing the protection of the server, right?